website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+                        self.loading = true;
+                        m.redraw();
+                        $('#rAFilleBar').replaceWith(
+                            '<div id="rAFilleBar" class="progress" style="height: 11px">' +
+                                '<div class="progress-bar progress-bar-success progress-bar-striped active" style="width:100%;"></div>' +
+                            '</div>'
+                        );
+                    },
+                    slide: function (){
+                        self.makeLine(canvas);
+                    },
+                    change: function (event, ui){
+                        self.loading = true;
+                        m.redraw();
+                        $('#rAFilleBar').replaceWith(
+                            '<div id="rAFilleBar" class="progress" style="height: 11px">' +
+                                '<div class="progress-bar progress-bar-success progress-bar-striped active" style="width:100%;"></div>' +
+                            '</div>'
+                        );
+                        self.page = 1;
+                        self.dateBegin = moment.utc(ui.values[0]*self.div);
+                        self.dateEnd = moment.utc(ui.values[1]*self.div);
+                        self.getLogs(false, true);
+                    }
+                });
+                $('#rASlider').slider('pips', {
+                    last: false,
+                    rest: 'label',
+                    step: self.steps,
+                    formatLabel: function(value){
+                        return String(moment.utc(value*self.div).format(self.formatPip));
+                    }
+                }).slider('float', {
+                    formatLabel: function(value){
+                        return String(moment.utc(value*self.div).format(self.formatFloat));
+                    }
+                });
+                self.getLogs(false, true);
+                var bar = $('#rASlider').find('.ui-slider-range');
+                bar.append(self.makeSliderProgress());
+                self.makeLine(canvas);
+            }
+            else {
+                //$('#rASlider').slider('option', 'values', values);
+                $('#rAFilleBar').replaceWith(self.makeSliderProgress());
+                self.makeLine(canvas);
+            }
+        };
+    },
+
+    view: function(ctrl, args){
+
+        var fileEvents = ((ctrl.eventNumbers.files/ctrl.totalEvents)*100 | 0) + (ctrl.eventNumbers.files ? 5 : 0);
+        var commentEvents = ((ctrl.eventNumbers.comments/ctrl.totalEvents)*100 | 0) + (ctrl.eventNumbers.comments ? 5 : 0);
+        var wikiEvents = ((ctrl.eventNumbers.wiki/ctrl.totalEvents)*100 | 0) + (ctrl.eventNumbers.wiki ? 5 : 0);
+        var nodeEvents = ((ctrl.eventNumbers.nodes/ctrl.totalEvents)*100 | 0) + (ctrl.eventNumbers.nodes ? 5 : 0);
+        var otherEvents = 100 - (fileEvents + commentEvents + wikiEvents + nodeEvents);
```

Comment:
Do these have to refresh at every view? Can you make a function to refresh the counts and then call it when it's needed?